src/data/validation/email.js

```diff
@@ -82,6 +82,33 @@ const validateEmailAddresses = (emails) => {
   return result;
 };
 
+// Each row in textarea or csv can contain email plus an optional salesforce id
+// Email and salesforce id will be separated by comma. This function will read
+// each row, split it by comma and then return an object with three properties:
+//      textEmails: All emails extracted from textarea
+//      CSVEmails: All emails extracted from CSV
+//      allEmails: Concatenation of `textEmails` and `CSVEmails`
+const extractEmails = (formData) => {
+  let textEmails = [];
+  let CSVEmails = [];
+  let allEmails = [];
+
+  if (formData[EMAIL_ADDRESS_TEXT_FORM_DATA] && formData[EMAIL_ADDRESS_TEXT_FORM_DATA].length) {
+    textEmails = formData[EMAIL_ADDRESS_TEXT_FORM_DATA].split(/\r\n|\n/).map(item => item.split(',')[0]);
+  }
+  if (formData[EMAIL_ADDRESS_CSV_FORM_DATA] && formData[EMAIL_ADDRESS_CSV_FORM_DATA].length) {
+    CSVEmails = formData[EMAIL_ADDRESS_CSV_FORM_DATA].map(item => item.split(',')[0]);
+  }
+
+  allEmails = [...textEmails, ...CSVEmails];
+
+  return {
+    textEmails: textEmails.length ? textEmails : undefined,
+    CSVEmails: CSVEmails.length ? CSVEmails : undefined,
+    allEmails,
```

Comment:
[suggestion] Given `allEmails` will always return an array, even when empty, I'm wondering if all the properties should follow that convention for consistency, which might simplify this logic a bit.

```js
return {
  textEmails,
  csvEmails,
  allEmails,
}
```

Reply:
Initially I was returning the exact same object as you suggested but [this line](https://github.com/openedx/frontend-app-admin-portal/pull/1034/files#diff-b9e9f5222df1e9f068d27a79a69afc3bed637207da72c9646d5be5b9b5586c74R125) fails when textEmails is [] and csvEmails have some emails. For example the below code evaluates to `[]` which is not what is expected.

```javascript
[] || ['abc@example.com']
```

This [test](https://github.com/openedx/frontend-app-admin-portal/blob/ed98264e122e47f80edaf6186a1e80c81892a1b7/src/data/validation/email.test.js#L74C1-L74C1) fails if we just return empty array.